ibc-apps/ics20-transfer/src/module.rs

```diff
@@ -203,11 +208,16 @@ pub fn on_acknowledgement_packet_validate<Ctx>(
 where
     Ctx: TokenTransferValidationContext,
 {
-    let data = serde_json::from_slice::<PacketData>(&packet.data)
-        .map_err(|_| TokenTransferError::FailedToDeserializePacketData)?;
+    let data = serde_json::from_slice::<PacketData>(&packet.data).map_err(|e| {
+        DecodingError::InvalidJson {
+            description: format!("failed to deserialize packet data: {e}"),
+        }
+    })?;
```

Comment:
I recalled that acknowledgement status failures are written into the host's state, and changing its message would be a state machine-breaking change. This means previously stored acknowledgements wouldn't be retrievable after the update. Additionally, the exact content of this message should be controlled and specified solely by the IBC application. However, the `InvalidJson` variant prepends 'invalid JSON data' by default, which might not be desirable for some apps, like ICA. Let's revert the error variant for places dealing with `AcknowledgementStatus` to the previous one. Sorry didn't bring this up earlier.

Reply:
So any of the InvalidJson variants with the description "failed to deserialize acknowledgment status" need to be reverted, but do the "failed to deserialize packet data" errors in the same on_acknowledgment_* functions also need to be reverted?

Reply:
Also, I just realized that the error message for the `FailedToDeserializeAck` variant has a miss-spelling. However, from what you said about how acknowledgments are stored in hosts' state, we can't actually fix this miss-spelling by changing the error message, right?

Reply:
Correct. More precisely both `on_recv_packet_execute` and `on_acknowledgement_packet_execute` should be reverted. In the former, the callback [returns an acknowledgement](https://github.com/cosmos/ibc-rs/blob/3ffdf2f1e5428a496d59e41117f1e75beea5c4c9/ibc-apps/ics20-transfer/src/module.rs#L173), and in the latter, it includes the acknowledgement [in the emitted event](https://github.com/cosmos/ibc-rs/blob/3ffdf2f1e5428a496d59e41117f1e75beea5c4c9/ibc-apps/ics20-transfer/src/module.rs#L273). So, any changes to either will cause breaking issues. 
In `on_acknowledgement_packet_validate`, it's not necessary, but for the sake of consistency better that be reverted too.

Reply:
> we can't actually fix this miss-spelling by changing the error message, right?

The `acknowledgement` you mean? I feel both w/ and w/o "e" should be correct. Though not sure. Anyway, let's keep it as before.

Reply:
The same functions need to be reverted in the same way in the analogous `NftTransferError` application as well, right?

Reply:
That's right.

Reply:
Made the changes in [283a606](https://github.com/cosmos/ibc-rs/pull/1325/commits/283a60620c5b0730be72e511a06631278d9b3d05), [3dc2ff5](https://github.com/cosmos/ibc-rs/pull/1325/commits/3dc2ff54458983d937c8e0dcca4b369649b51ec0), and [94b12af](https://github.com/cosmos/ibc-rs/pull/1325/commits/94b12affcc80762e965e4f79ff96580198f73f97).